### ISSUE TITLE
Rename by symbol using F2

### ISSUE THREAD

**Issue by aberba** [Pure-D/code-d#316]
Its quite handy to have `F2` work for renaming a variable or function across all areas its being used. This usually works when the variable or function is selected.

**Comment by jaens** [comment]
This works (somewhat) in https://github.com/d-language-server/dls

The implementation looks pretty uncomplicated, it just does a "find all references" for the symbol, and then textually replaces all the references with the new symbol.

**Comment by WebFreak001** [comment]
yes, find all references isn't yet implemented in serve-d yet, so once that's implemented a few related features should be easier to implement
